### PR TITLE
Ensure initialWindowMetrics is called on DispatchQueue.main

### PR DESCRIPTION
**Pull request by hakonk**
# Why

When running Expo bare, the Main thread checker complained. It turned out that `initialWindowMetrics` was called on another thread than main.

# How

Dispatch sync on main if `!Thread.isMainThread`. Additionally, `RNCSafeAreaProviderManager` is made final.

# Test Plan

Only manually verified that the main thread checker does not complain when the Expo bare app runs.

# Checklist

There is no change to the public API, and therefore I believe there is no need to document this further, neither on the website nor the README. Please let me know if this is not the right assumption.

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-menu/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 ### 🎉 New features
 
 ### 🐛 Bug fixes
+- [iOS] Fixed issue where `UIWindow` was called on off the main thread. ([#29559](https://github.com/expo/expo/pull/29559) by [@hakonk](https://github.com/hakonk))
```
Comment (lukmccall) [inline packages/expo-dev-menu/CHANGELOG.md:10]:
```suggestion

- [iOS] Fixed issue where `UIWindow` was called on off the main thread. ([#29559](https://github.com/expo/expo/pull/29559) by [@hakonk](https://github.com/hakonk))
```

File: packages/expo-dev-menu/vendored/react-native-safe-area-context/ios/SafeAreaView/DevMenuRNCSafeAreaProviderManager.swift [hunk 2]
```diff
@@ -13,19 +13,36 @@ public class RNCSafeAreaProviderManager: Module {
...
   private var initialWindowMetrics: [String: Any]? {
-    guard let window = UIApplication.shared.keyWindow else {
-      return [:]
-    }
+    syncOnMain {
+      guard let window = UIApplication.shared.keyWindow else {
+        return [:]
+      }
+
+      if #available(iOS 11.0, *) {
```
Comment (lukmccall) [inline packages/expo-dev-menu/vendored/react-native-safe-area-context/ios/SafeAreaView/DevMenuRNCSafeAreaProviderManager.swift:21]:
```suggestion
```
We don't need that `if` here. We don't support iOS below 11 anymore.

**Review by lukmccall — COMMENTED** [review]
It's looking good, but we don't need to handle case where iOS version is `< 11`.

File: react-native-lab/react-native
Comment (lukmccall) [inline react-native-lab/react-native:1]: Can you revert that change?

Comment (hakonk) [reply react-native-lab/react-native:1]: I seem to be doing something wrong here. What is the standard procedure for reverting changes to the submodule `react-native-labs/react-native`?

File: packages/expo-dev-menu/vendored/react-native-safe-area-context/ios/SafeAreaView/DevMenuRNCSafeAreaProviderManager.swift [hunk 3]
```diff
@@ -13,25 +13,35 @@ public class RNCSafeAreaProviderManager: Module {
...
+          "bottom": safeAreaInsets.bottom,
+          "left": safeAreaInsets.left
+        ],
+        "frame": [
+          "x": window.frame.origin.x,
+          "y": window.frame.origin.y,
+          "width": window.frame.size.width,
+          "height": window.frame.size.height
+        ]
       ]
```
Comment (lukmccall) [inline packages/expo-dev-menu/vendored/react-native-safe-area-context/ios/SafeAreaView/DevMenuRNCSafeAreaProviderManager.swift:36]:
```suggestion
        ]
      ]
    }
```

**Review by lukmccall — APPROVED** [review]
Looking good!

**Pull request merged**